control/Director.php

```diff
@@ -384,6 +384,7 @@ protected static function handleRequest(SS_HTTPRequest $request, Session $sessio
 					try {
 						$result = $controllerObj->handleRequest($request, $model);
 					} catch(SS_HTTPResponse_Exception $responseException) {
+						HTTPCacheControl::singleton()->disableCache(true);
```

Comment:
Question: Should we cache 404s? It feels like it's better to leave this up to usercode than enforce here (especially with force = true making it hard to reverse).

Reply:
Hrm, good point. Perhaps this should be disableCache(false) with more disableCache(true) only used prior to the throwing of more specific responses.

Reply:
disableCache(false) is the same priority of caching as "never cache if there's a session". blocking cache on 404s, 302s, 301s, etc in such circumstance seems okay.